test/integration/serverless/log_normalize.py

```diff
@@ -0,0 +1,148 @@
+import argparse
+import json
+import re
+import sys
+
+def normalize_metrics(stage):
+    return [
+            replace(r'raise Exception', r'\n'),
+            require(r'BEGINMETRIC.*ENDMETRIC'),
+            exclude(r'BEGINMETRIC'),
+            exclude(r'ENDMETRIC'),
+            replace(r'(ts":)[0-9]{10}', r'\1XXX'),
+            replace(r'(min":)[0-9\.e\-]{1,30}', r'\1XXX'),
+            replace(r'(max":)[0-9\.e\-]{1,30}', r'\1XXX'),
+            replace(r'(cnt":)[0-9\.e\-]{1,30}', r'\1XXX'),
+            replace(r'(avg":)[0-9\.e\-]{1,30}', r'\1XXX'),
+            replace(r'(sum":)[0-9\.e\-]{1,30}', r'\1XXX'),
+            replace(r'(k":\[)[0-9\.e\-]{1,30}', r'\1XXX'),
+            replace(r'(datadog-nodev)[0-9]+\.[0-9]+\.[0-9]+', r'\1X.X.X'),
+            replace(r'(datadog_lambda:v)[0-9]+\.[0-9]+\.[0-9]+', r'\1X.X.X'),
+            replace(r'dd_lambda_layer:datadog-go[0-9.]{1,}', r'dd_lambda_layer:datadog-gox.x.x'),
+            replace(r'(dd_lambda_layer:datadog-python)[0-9_]+\.[0-9]+\.[0-9]+', r'\1X.X.X'),
+            replace(r'(serverless.lambda-extension.integration-test.count)[0-9\.]+', r'\1'),
+            replace(r'(architecture:)(x86_64|arm64)', r'\1XXX'),
+            replace(stage, 'XXXXXX'),
+            exclude(r'[ ]$'),
+    ]
+
+def normalize_logs(stage):
+    return [
+            require(r'BEGINLOG.*ENDLOG'),
+            exclude(r'BEGINLOG'),
+            exclude(r'ENDLOG'),
+            replace(r'("timestamp":\s*?)\d{13}', r'\1"XXX"'),
+            replace(r'\d{4}-\d{2}-\d{2}\s\d{2}:\d{2}:\d{2}:\d{3}', 'TIMESTAMP'),
+            replace(r'\d{4}-\d{2}-\d{2}T\d{2}:\d{2}:\d{2}\.\d{3}Z', 'TIMESTAMP'),
+            replace(r'\d{4}\/\d{2}\/\d{2}\s\d{2}:\d{2}:\d{2}', 'TIMESTAMP'),
+            replace(r'\d{4}-\d{2}-\d{2}\s\d{2}:\d{2}:\d{2}', 'TIMESTAMP'),
+            replace(r'([a-zA-Z0-9]{8}-[a-zA-Z0-9]{4}-[a-zA-Z0-9]{4}-[a-zA-Z0-9]{4}-[a-zA-Z0-9]{12})', r'XXX'),
+            replace(stage, 'XXXXXX'),
+            replace(r'(architecture:)(x86_64|arm64)', r'\1XXX'),
+            sort_by(lambda log: log["message"]["message"]),
+            # ignore a Lambda error that occurs sporadically for log-csharp see here for more info:
+            # https://repost.aws/questions/QUq2OfIFUNTCyCKsChfJLr5w/lambda-function-working-locally-but-crashing-on-aws
+            # TODO
+            # perl -n -e "print unless /LAMBDA_RUNTIME Failed to get next invocation. No Response from endpoint/ or \
+            #  /An error occurred while attempting to execute your code.: LambdaException/ or \
+            #  /terminate called after throwing an instance of 'std::logic_error'/ or \
+            #  /basic_string::_M_construct null not valid/" |
```

Comment:
This `perl` statement was present in the origin bash version. I've left it commented out because in all my testing, I did not see it actually make any changes to the logs we receive.